### ISSUE TITLE
Decide how we should handle showing upvotes and downvotes to the user

### ISSUE THREAD

**Issue by nickserv** [toroidal-code/codeflash#15]

**Comment by kristenmills** [comment]
As opposed to?

**Comment by nickserv** [comment]
Here are some of my ideas:
- upvotes only, upvotes are displayed (likes basically)
- upvotes and downvotes, total is displayed (like Stack Overflow)
- upvotes and downvotes, total is displayed if the post is not extremely new (like Reddit)
- upvotes and downvotes, both are displayed (like Reddit with RES)

**Comment by nickserv** [comment]
Also, there's the question of whether you should be able to see who voted and how they voted.

**Comment by smy2748** [comment]
I'd say that we should display the total.  We could display the upvotes and downvotes as well, I suppose.  I'm not sure which one would be more effective at dispelling groupthink and encourage the community to vote based on how they feel.  

Perhaps we could not show the point total of a solution until after they've voted.  We could still keep things in order, but not show how far apart they are.

**Comment by nickserv** [comment]
I feel like scoring would be more fair if we went with the Reddit-ish ideas. And that's a cool idea for keeping votes fair, but what if someone just wants to see exactly how well a solution is graded without voting on it?

**Comment by StacyJCarlos** [comment]
I think we should use a weighted vote system. That way people who have earned more credibility  will be able to count for more ups or downs and of course if this would allow for admin votes to have a little bit more authority.

**Comment by nickserv** [comment]
I like that idea, but we need to make sure that this is fair enough so that votes of newer users still count enough. We don't want voting power to snowball with the users with the most credibility. In other words, it should be easy for new users to eventually get lots of cred if they do well enough.

**Comment by smy2748** [comment]
Perhaps we could keep upvotes and downvotes "invisible" for a certain amount of time directly after a solution has been submitted and then allow people to see them.  Or we could display just the percentage of approval.

We could scale the voting based on the userbase and the number of points that a user has already.  The issue with scaling is that we'd be introducing a variable amount of points added with up or down votes and would need to keep track of which users voted on which solution.

**Comment by StacyJCarlos** [comment]
There should definitely be a seniority system concerning votes so that the people have been using the site for extended periods of time would have a much more valuable vote than those who have just joined the site. 

The downside would be that people would be slightly put off by this system, but overall I think it makes votes a more accurate way of rating code.

**Comment by nickserv** [comment]
@smy2748 I like the idea of making upvotes and downvotes temporarily invisible for new solutions.

@smy2748 @StacyJCarlos I disagree, I think power would snowball with users who already have the most power. Sites like Reddit and Stack Overflow give all users the same voting power, and I think we should stick with that.

**Comment by nickserv** [comment]
2 people are for seniority voting, and 3 people are against it.

As a result, we will not do seniority voting, and all votes from all users will be worth 1 point (with a max of one vote per solution).

**Comment by nickserv** [comment]
So, based on our previous discussions and what we have tended to agree on, I propose these ideas for handling votes:
- In the UI, display total votes along with upvote/downvote options.
- Make vote totals invisible temporarily on new posts.
- Every vote counts as one point (up or down), and users can only vote once per post.

Notes:
- When I say posts, I mean solutions and anything that will need votes, such as problems.
- This voting system will pretty much work like Reddit.

Please tell me if you agree with this. If I get two more agrees (I count as one), these ideas will have the majority vote, and we will be using them.

**Comment by kristenmills** [comment]
agreed

**Comment by StacyJCarlos** [comment]
I like the not displaying votes for new posts and I like the display of the up votes down vote, but I still believe that there should be a way to allow people who have consistently used the site to have more of a say than people who have just joined. I have though t of a way to implement that as well.

Instead of having a seniority system that people may not like, we could implement a vote 'booster' to people that people would receive for being selected as the winning implementation. They could use this only once and it would be counted as maybe 5 votes or less depending what we would like. What do we think?

**Comment by toroidal-code** [comment]
no boosters. keep it simple.

**Comment by StacyJCarlos** [comment]
Okay 'boosters' may be a little extreme, but a vote 'multivitamin' which would simply make votes more valid.

**Comment by nickserv** [comment]
Yingling agrees with my ideas. He said that he might want something like vote boosters in the future, but that we should stick to simple 1 point per vote stuff for now.

As a result, we now have a majority consensus. [These ideas](https://github.com/codeflash/codeflash/issues/15#issuecomment-13341806) will be implemented, and this issue is done for now.

**Comment by smy2748** [comment]
It's been a year since the "vote booster" idea was postponed.  What are our thought now?
